### PR TITLE
feat(suite): stop coinjoin

### LINKED ISSUE #7871: Use `Stop` instead of distinguishing `Pause` and `Cancel` in coinjoin account
**Why?**
During testing, it turns out that users don't understand the difference between `pause` and `cancel` in our UI. 

**Current behaviour**
`Pause` - Pauses the coinjoin, but the user can resume at any time with one click. _Cannot change conjoin parameters like `desired anonymity level` or `max mining fee`_
`Cancel` - Cancels the coinjoin. User must reconfirm terms with new coinjoin including reconfirmation of the total number of signatures on the Trezor. 

<details>
  <summary> Pause state is sometimes triggered automatically</summary>

 - entering Send section (continues after leaving the section)
 - losing internet/Tor connection (continues after reconnection)
 - turning off Suite (doesn't start automatically)
 
</details>

### Proposal

 - If the user stops the coinjoin
    - ~~He can resume coinjoin again with one click (as before pause)~~
    - Once user presses the play button he will have to confirm the terms and the new number of rounds again (until we implement: https://github.com/trezor/trezor-suite/issues/7456
    - Only way how to reach `pause` state (and automatically resume) is when user access the send page. And only until we solve it by https://github.com/trezor/trezor-suite/issues/8373

 - Rename feature to `Stop`
 - Remove `...` 
 
Screenshot 2023-03-20 at 14 59 04

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* remove an ability to manually pause Coinjoin
* removed actions menu next to the CJ status wheel
* clicking on the wheel now stops CJ
* the button in the critical phase modal now stops a session after a current round instead of pausing
* added a stop button when a session is interrupted

Optionally I propose renaming "pause" to "interrupted" in both UI and code to better reflect its new meaning.

## Related Issue

Resolve #7871

## Screenshots:

image
image
image

### CODE REVIEW

File: packages/suite/src/actions/wallet/coinjoinAccountActions.ts [hunk 1]
```diff
@@ -164,12 +164,12 @@ const coinjoinSessionStarting = (accountKey: string, isStarting: boolean) =>
...
 
-const coinjoinSessionAutopause = (accountKey: string, isAutopaused: boolean) =>
+const coinjoinSessionAutostop = (accountKey: string, isAutostoped: boolean) =>
     ({
-        type: COINJOIN.SESSION_AUTOPAUSE,
+        type: COINJOIN.SESSION_AUTOSTOP,
         payload: {
             accountKey,
-            isAutopaused,
+            isAutostoped,
```
Comment (komret) [inline packages/suite/src/actions/wallet/coinjoinAccountActions.ts]: typo: should be 'isAutostopped'

File: packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts [hunk 2]
```diff
@@ -135,8 +135,9 @@ const RESTORE_SESSION_B_ACTIONS = [
 
 export const fixtures = [
     {
-        description: 'pause coinjoin session when remembered device disconnects',
+        description: 'stop coinjoin session stops remembered device disconnects',
```
Comment (komret) [inline packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts]: typo

File: packages/suite/src/utils/suite/storage.ts [hunk 3]
```diff
@@ -31,19 +31,7 @@ export const serializeDevice = (device: AcquiredDevice, forceRemember?: true) =>
  */
 export const serializeCoinjoinSession = (coinjoinAccount: CoinjoinAccount) => {
```
Comment (komret) [inline packages/suite/src/utils/suite/storage.ts]: Let's refactor this a bit:

```ts
export const serializeCoinjoinSession = (coinjoinAccount: CoinjoinAccount) => {
    const { session, transactionCandidates, ...propertiesToSave } = coinjoinAccount;
    return propertiesToSave;
};
```

If you go for this, please also adjust the comment and remove this line from `saveCoinjoinAccount`:
`delete serializedAccount.transactionCandidates; // do not store tx candidates`

File: packages/suite/src/types/wallet/coinjoin.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ export interface CoinjoinSession extends CoinjoinSessionParameters {
     timeCreated: number; // timestamp when was created
     timeEnded?: number; // timestamp when was finished
     paused?: boolean; // current state
-    isAutoPauseEnabled?: boolean; // auto pause after current round
+    isAutoStopEnabled?: boolean; // auto pause after current round
     interrupted?: boolean; // it was paused by force
```
Comment (komret) [inline packages/suite/src/types/wallet/coinjoin.ts]: Now that coinjoin cannot be paused manually, `interrupted` always equals `paused`. You can remove the `interrupted` param.

File: packages/components/src/images/icons/stop.svg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (komret) [inline packages/components/src/images/icons/stop.svg]: Here's a simplified version that will also look better in Storybook and Finder:
```
<svg viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg">
<path id="Vector" d="M25.9574 4H6.04255C5.50083 4 4.9813 4.21071 4.59825 4.58579C4.2152 4.96086 4 5.46957 4 6V26C4 26.5304 4.2152 27.0391 4.59825 27.4142C4.9813 27.7893 5.50083 28 6.04255 28H25.9574C26.4992 28 27.0187 27.7893 27.4017 27.4142C27.7848 27.0391 28 26.5304 28 26V6C28 5.46957 27.7848 4.96086 27.4017 4.58579C27.0187 4.21071 26.4992 4 25.9574 4Z" />
</svg>
```

**Review by komret — CHANGES_REQUESTED** [review]
I like the simplification! Agreed that `interrupted` is clearer than `paused`, but the word is a bit long to fit into the circle.

File: packages/suite/src/types/wallet/coinjoin.ts (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/types/wallet/coinjoin.ts]: i was also considering renaming `paused` to `interrupted`, what would you say?

Comment (komret) [reply packages/suite/src/types/wallet/coinjoin.ts]: It is more precise, but I'd stick to whatever the UI is showing.

File: packages/suite/src/actions/wallet/coinjoinAccountActions.ts (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/actions/wallet/coinjoinAccountActions.ts:172]: ebac9b502cc404e3706340bcd9f578fdb95590c9

File: packages/suite/src/utils/suite/storage.ts (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/utils/suite/storage.ts]: fbb5905c58538397db0bf98750fe920977060beb

File: packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts]: 1475535cf81448e8370ca4e3ba7398137322b06d

File: packages/suite/src/types/wallet/coinjoin.ts (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/types/wallet/coinjoin.ts]: 02d04c49df70db4f476198875016885bab17a940

File: packages/components/src/images/icons/stop.svg (hunk 5 above)
Comment (dahaca) [reply packages/components/src/images/icons/stop.svg]: 70f0c624a376fc25afb6a1bcaf986836cb46e00a

File: packages/suite/src/actions/wallet/coinjoinAccountActions.ts (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/actions/wallet/coinjoinAccountActions.ts]: 926731886a9e40383cdcfa7b9c5e41dc8e4e57d8

File: packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts (hunk 2 above)
Comment (komret) [reply packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts]: I still don't understand the sentence :D Shouldn't it say "stop coinjoin session **when** remembered device disconnects"?

File: packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts [hunk 6]
```diff
@@ -276,14 +274,14 @@ export const coinjoinMiddleware =
 
         if (action.type === COINJOIN.CLIENT_SESSION_PHASE) {
             const { accountKeys } = action.payload;
-            const isAlredyInterrupted = api
+            const isAlredyPaused = api
```
Comment (komret) [inline packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts]: Please fix the typo here: `isAlreadyPaused`

Comment (dahaca) [reply packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts]: done

File: packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/middlewares/wallet/__fixtures__/coinjoinMiddleware.ts]: of course....

**Review by komret — APPROVED** [review]

**Pull request merged**